### ISSUE TITLE
Update `linuxkit/extend` to support multiple partitions

### ISSUE THREAD

**Issue by ernoaapa** [linuxkit/linuxkit#2971]
Hi!
I have been playing around with Linuxkit with RaspberryPI and now I'm writing an example how to use persisted disk with RaspberryPI ( more in #2964 )

I would like to support two partitions, first for booting and second as persisted storage.
At the moment `linuxkit/extend` supports only one partition, but it would be really cool to be able to create one linuxkit build which extends the second partition up to the max so I could create single img-file what works in different size of sd-cards.

I can provide PR but before that would like to hear some feedback what would be the best way to implement it. Following would be enough for my use case:
- Update it to extend always the last partition
- Add support for passing partition name as an argument. E.g. `/dev/mmcblk0p2` and it resolves that it's the `/dev/mmcblk0` disk second partition.
- Add optional `--partitionIndex` flag

What do you think?

**Description**

**Steps to reproduce the issue:**

**Describe the results you received:**

**Describe the results you expected:**

**Additional information you deem important (e.g. issue happens only occasionally):**

**Comment by justincormack** [comment]
I think probably always extending the last partition is fine.

**Comment by rn** [comment]
yes extending the last partition would be ok, also taking an argument like the `format` package does would be nice.

In general, I think eventually we want to switch over the rootfs to squashfs (not just for the RPi3) so that it doesn't need to run out of ram disk and consume memory

**Comment by ernoaapa** [comment]
Thanks for giving feedback.

Actually after testing the "extend the last partition", faced one problem. There are basically two dirs what I want to be mounted into the physical disk to avoid running out of memory in RPI3, `/var` and `/containers`.
Saw @justincormack ticket https://github.com/linuxkit/linuxkit/issues/1285 about taking the `/containers` under `/var`. Do you know why `/containers` is not under `/var`? Probably there's good reason for that :)

If you have any good ideas, how to mount both dirs to the single partition, I happily hear that :)

**Comment by justincormack** [comment]
To fix the memory issue we need to add support for not using initramfs as the boot for disk based images. We are planning to add squashfs support for this, in which case their should not be a memory issue.

**Comment by justincormack** [comment]
See also https://github.com/zededa/zenbuild/pull/21

**Comment by ernoaapa** [comment]
Ok, using squashfs sounds interesting. Any progress regarding getting support into linuxkit?

I'm not familiar with squashfs so it's not clear to me, would the root still be in memory but there would be just more free memory or would it support e.g. using another ext4 partition for persistence?
I.e. if I have raspi with 1G memory, but 32G sd-card, could I download large/multiple containers because it would use the sd-card for persisting the images, etc.?

Thanks for all the help and info.

**Comment by justincormack** [comment]
Squashfs is another filesystem (but read only, compressed). It would be on its own partition and not be in memory.

**Comment by rn** [comment]
https://github.com/linuxkit/linuxkit/pull/3002 adds support for building and booting with squashfs based root filesystem. This isn't quite what you need to the RPi3 but a start. Basically for the RPi3 I think you'd need to create a partition for uboot with the kernel and another partition with the squashfs based rootfs

**Comment by ernoaapa** [comment]
@rn creating the partitions is not a problem. At the moment I'm adding a feature to the [linuxkit-server](https://github.com/ernoaapa/linuxkit-server) that you can request the linuxkit build in `img` format and define desired partitions. That would be writable to sd-card directly with for example `dd`.
I'll take a look at the #3002, hopefully after that, I can really start running linuxkit in RPI3, yay! 😃Thanks!
